bulldozer/signals.go

```diff
@@ -21,109 +21,322 @@ import (
 			}
 		}
 	}
 
+	return descriptions, matches, nil
+}
+
+// commentMatches determines which comment signals match the given PR. It returns:
+// - A list of descriptions for each matched signal
+// - A list of the matched signals
+func (s *Signals) commentMatches(ctx context.Context, pullCtx pull.Context, tag string) ([]string, Signal, error) {
+	logger := zerolog.Ctx(ctx)
+	matches := Signal{}
+	descriptions := []string{}
+
 	body := pullCtx.Body()
 	comments, err := pullCtx.Comments(ctx)
 	if err != nil {
-		return false, "unable to list pull request comments", err
+		return descriptions, matches, errors.Wrap(err, "unable to list pull request comments")
 	}
 
 	if len(comments) == 0 {
 		logger.Debug().Msgf("No comments found to match against")
 	}
+
 	for _, signalComment := range s.Comments {
 		if body == signalComment {
-			return true, fmt.Sprintf("pull request body is a %s comment: %q", tag, signalComment), nil
-		}
-		for _, comment := range comments {
-			if comment == signalComment {
-				return true, fmt.Sprintf("pull request has a %s comment: %q", tag, signalComment), nil
+			matches = append(matches, signalComment)
+			descriptions = append(descriptions, fmt.Sprintf("pull request body is a %s comment: %q", tag, signalComment))
+		} else {
+			// TODO: Use a set and remove the else
+			for _, comment := range comments {
+				if comment == signalComment {
+					matches = append(matches, signalComment)
+					descriptions = append(descriptions, fmt.Sprintf("pull request has a %s comment: %q", tag, signalComment))
+				}
 			}
 		}
+
+	}
+
+	return descriptions, matches, nil
+}
+
+// commentSubstringMatches determines which comment substring signals match the given PR. It returns:
+// - A list of descriptions for each matched signal
+// - A list of the matched signals
+func (s *Signals) commentSubstringMatches(ctx context.Context, pullCtx pull.Context, tag string) ([]string, Signal, error) {
+	logger := zerolog.Ctx(ctx)
+	matches := Signal{}
+	descriptions := []string{}
+
+	body := pullCtx.Body()
+	comments, err := pullCtx.Comments(ctx)
+	if err != nil {
+		return descriptions, matches, errors.Wrap(err, "unable to list pull request comments")
 	}
 
 	if len(s.CommentSubstrings) == 0 {
 		logger.Debug().Msgf("No comment substrings found to match against")
 	}
+
 	for _, signalSubstring := range s.CommentSubstrings {
 		if strings.Contains(body, signalSubstring) {
-			return true, fmt.Sprintf("pull request body matches a %s substring: %q", tag, signalSubstring), nil
-		}
-		for _, comment := range comments {
-			if strings.Contains(comment, signalSubstring) {
-				return true, fmt.Sprintf("pull request comment matches a %s substring: %q", tag, signalSubstring), nil
+			matches = append(matches, signalSubstring)
+			descriptions = append(descriptions, fmt.Sprintf("pull request body matches a %s substring: %q", tag, signalSubstring))
+		} else {
+			// TODO: Use a set and remove the else
```

Comment:
Are you thinking about something like:
```
allStrings := make(map[string][]string) {
"body" : []string{body},
"comments": comments,
}

for k, v := range allStrings {
  if strings.Contains(v, signalSubstring) {
  ...
}
```

Reply:
I think we'd want to do:
```golang
allStrings := make(map[string][]string) { 
			"body" : []string{body}, 
			"comments": comments, 
		} 
		
		for label, strings := range allStrings { 
			for _, s := range strings {
				if s == signalComment {
					matches = append(matches, signalComment)
					descriptions = append(descriptions, fmt.Sprintf("pull request has a %s %s: %q", tag, label, signalComment))
				}
			}
		}
```

Which seems a little harder to understand than the original. Going to just remove the TODO